src/screens/ConfirmEmail.js

```diff
@@ -0,0 +1,190 @@
+import React, {useEffect, useState} from 'react';
+import {
+  SafeAreaView,
+  View,
+  ScrollView,
+  Text,
+  TouchableOpacity,
+  StyleSheet,
+  Alert,
+} from 'react-native';
+import {Colors} from '../utils/Colors';
+import {NEXT_MESSAGE} from '../utils/Constants';
+import CodeConfirmTxtInput from '../components/CodeConfirmTxtInput';
+
+const ConfirmEmail = ({route}) => {
+  const {usersEmail} = route.params;
+  let codeConfirmation = ['1', '2', '3', '4'];
+  let auxiliarCounter = 0;
+  const [formData, setFormData] = useState({
+    fieldOne: '',
+    fieldTwo: '',
+    fieldThree: '',
+    fieldFour: '',
+  });
+
+  useEffect(() => {
+    setFormData({
+      fieldOne: '',
+      fieldTwo: '',
+      fieldThree: '',
+      fieldFour: '',
+    });
+  }, []);
+
+  const handleOnChange = ({number, id}) => {
+    setFormData({
+      ...formData,
+      [id]: number,
+    });
+  };
+
+  const resetForm = () => {
+    setFormData({
+      fieldOne: '',
+      fieldTwo: '',
+      fieldThree: '',
+      fieldFour: '',
+    });
+  };
+
+  const checkCodeConfirmation = () => {
+    if (
+      formData.fieldOne === '' &&
+      formData.fieldTwo === '' &&
+      formData.fieldThree === '' &&
+      formData.fieldFour === ''
+    ) {
+      Alert.alert('type all the code');
+    } else {
+      let userCodeConfirmation = [
+        formData.fieldOne,
+        formData.fieldTwo,
+        formData.fieldThree,
+        formData.fieldFour,
+      ];
```

Comment:
You could use js destructuration
```suggestion
      let userCodeConfirmation = [
        ...formData
      ];
```